torch/distributed/distributed_c10d.py

```diff
@@ -271,6 +271,14 @@ def _rank_not_in_group(group: ProcessGroup):
     return group == GroupMember.NON_GROUP_MEMBER
 
 
+def _warn_not_in_group(op_name):
+    global_rank = -1 if GroupMember.WORLD is None else GroupMember.WORLD.rank()
```

Comment:
GroupMember.WORLD = None should never happen right, since we always expect to have initialized a default PG? Should we throw in that case?

Reply:
I assume the following case is allowed without calling `init_process_group`?

```python
pg = ProcessGroupGloo(...)
dist.all_reduce(tensor, group=pg)
```